test/index.test.js

```diff
@@ -911,5 +913,73 @@ describe('mongoose module:', function() {
       const m = new mongoose.Mongoose();
       assert.deepEqual(await m.syncIndexes(), {});
     });
+    describe('global `allowDiskUse` (gh-11478)', () => {
+      this.afterEach(() => sinon.restore());
+
+      it('is `undefined` by default', async() => {
+        // Arrange
+        const m = new mongoose.Mongoose();
+
+        const db = await m.connect(start.uri);
+
+        const userSchema = new m.Schema({
+          name: String
+        });
+
+        const User = db.model('User', userSchema);
+
+        const nativeAggregateSpy = sinon.spy(User.collection, 'aggregate');
+
+        // Act
+        await User.aggregate([{ $match: {} }]);
+
+        // Assert
+        const optionsSentToMongo = nativeAggregateSpy.args[0][1];
+        assert.strictEqual(optionsSentToMongo.allowDiskUse, undefined);
+      });
+
+      it('works when set to `true` and no option provided', async() => {
+        // Arrange
+        const m = new mongoose.Mongoose();
+        m.set('allowDiskUse', true);
+
+        const db = await m.connect(start.uri);
+
+        const userSchema = new m.Schema({
+          name: String
+        });
+
+        const User = db.model('User', userSchema);
+        const nativeAggregateSpy = sinon.spy(User.collection, 'aggregate');
```

Comment:
Please try to avoid this. It is good to not restore the spy in the it-Block as a failing test could mean that a spy is not restored. So restoring in an after block is recommended.

Do in the describe-block after line 916 a let nativeAggregateSpy and assign a value to it in this line (=> remove the const here)

In the after block you do just the 
nativeAggregateSpy && nativeAggregateSpy.restore()

I had cases where I used a global spy or wanted to introduce a global spy and was massively blocked because there was general sinon.restore() used.

Reply:
I want to restore everything to its original state after each test.
I learned the hard way that trying to control things globally introduces more problems that are harder to fix in the long run.

You can have the best of both worlds by grouping up your global fakes into a function(s) and then explicitly calling them at the beginning of the `it` block. That way we won't have to jump back and forth in the codebase trying to figure out what caused some specific behavior.